src/internal/m365/service/exchange/export_test.go

```diff
@@ -0,0 +1,433 @@
+	require.NoError(suite.T(), err, "read email file")
+
+	table := []struct {
+		name              string
+		version           int
+		backingCollection data.RestoreCollection
+		expectedItems     []export.Item
+	}{
+		{
+			name:    "single item",
+			version: 1,
+			backingCollection: data.NoFetchRestoreCollection{
+				Collection: dataMock.Collection{
+					ItemData: []data.Item{
+						&dataMock.Item{
+							ItemID: "id1",
+							Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+						},
+					},
+				},
+			},
+			expectedItems: []export.Item{
+				{
+					ID:   "id1",
+					Name: "id1.eml",
+					Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+				},
+			},
+		},
+		{
+			name:    "multiple items",
+			version: 1,
+			backingCollection: data.NoFetchRestoreCollection{
+				Collection: dataMock.Collection{
+					ItemData: []data.Item{
+						&dataMock.Item{
+							ItemID: "id1",
+							Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+						},
+						&dataMock.Item{
+							ItemID: "id2",
+							Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+						},
+					},
+				},
+			},
+			expectedItems: []export.Item{
+				{
+					ID:   "id1",
+					Name: "id1.eml",
+					Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+				},
+				{
+					ID:   "id2",
+					Name: "id2.eml",
+					Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+				},
+			},
+		},
+		{
+			name:    "items with success and fetch error",
+			version: version.Groups9Update,
+			backingCollection: data.FetchRestoreCollection{
+				Collection: dataMock.Collection{
+					ItemData: []data.Item{
+						&dataMock.Item{
+							ItemID: "id0",
+							Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+						},
+						&dataMock.Item{
+							ItemID:  "id1",
+							ReadErr: assert.AnError,
+						},
+						&dataMock.Item{
+							ItemID: "id2",
+							Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+						},
+					},
+				},
+			},
+			expectedItems: []export.Item{
+				{
+					ID:   "id0",
+					Name: "id0.eml",
+					Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+				},
+				{
+					ID:   "id2",
+					Name: "id2.eml",
+					Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+				},
+				{
+					ID:    "",
+					Error: assert.AnError,
+				},
+			},
+		},
+	}
+
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			stats := data.ExportStats{}
+			ec := exchange.NewExportCollection(
+				"",
+				[]data.RestoreCollection{test.backingCollection},
+				test.version,
+				&stats)
+
+			items := ec.Items(ctx)
+
+			count := 0
+			size := 0
+			fitems := []export.Item{}
+
+			for item := range items {
+				if item.Error == nil {
+					count++
+				}
+
+				if item.Body != nil {
+					b, err := io.ReadAll(item.Body)
+					assert.NoError(t, err, clues.ToCore(err))
+
+					size += len(b)
+					item.Body = io.NopCloser(bytes.NewBuffer(b))
```

Comment:
nit: do we need to make the body available again for this test or can we remove this line?